ui/app/controllers/vault/cluster/access/oidc/clients/client.js

```diff
@@ -0,0 +1,18 @@
+import Controller from '@ember/controller';
+import { inject as service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+
+export default class OidcClientController extends Controller {
+  @service router;
+  @tracked showHeader = true;
+
+  constructor() {
+    super(...arguments);
+    this.router.on('routeDidChange', (transition) => this.showOrHideHeader(transition));
+  }
+
+  showOrHideHeader({ targetName }) {
+    // hide header when rendering the edit form (client-form component has separate header)
+    this.showHeader = targetName.includes('edit') ? false : true;
```

Comment:
Instead of making this a tracked property, could we have `showHeader` be a getter that returns whether the target name includes edit? That would be more stateful and wouldn't rely on transition actions

Reply:
Funny you suggest this 😅 - I had it as a getter before and it wasn't re-firing when navigating between routes. But I was likely doing something wrong...will try again!

Reply:
![controller-fire](https://user-images.githubusercontent.com/68122737/182257819-b24ca1c5-38a3-4f82-a5b8-e19a574bf45a.gif)

okay so the controller here only fires on two occasions:
1. **initial** transition to the `/oidc/clients/:name/details` 
2. anytime we transition to `/oidc/clients/:name/providers` 

I refactored it using a getter, following how Jordan set up the `OidcConfigureController` - but not sure if this is what you meant!